src/libraries/System.Collections/tests/Generic/PriorityQueue/PriorityQueue.PropertyTests.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using FsCheck;
+using FsCheck.Xunit;
+using Xunit;
+
+namespace System.Collections.Tests
+{
+    public static class PriorityQueue_PropertyTests
+    {
+        const int MaxTest = 100;
+
+        private readonly static IComparer<string> s_stringComparer = StringComparer.Ordinal;
+        
+        [Property(MaxTest = MaxTest)]
```

Comment:
I haven't seen this construct in our tests before.  What is this doing?

Reply:
It's effectively a theory that gets fed pseudo-randomized inputs. Essentially a structured fuzzer, without being coverage guided.

Reply:
I would prefer we not use this in any of our functional tests. If we want a dedicated fuzziness site that runs separately,  fine.

It's yet one more style of test devs need to understand.  It doesn't have the various workarounds devs know to use of something needs to be disabled, e.g. conditional. It's hard to reason about exactly what this is testing,  predictability,  and execution time. It adds another dependency to the build.  Etc.

Reply:
I've replaced the Property attributes with regular theories that manually generate random inputs.

Reply:
Thank you :)